activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -0,0 +1,389 @@
+          expires_in: expires_in, race_condition_ttl: race_condition_ttl
+      end
+
+      def redis
+        @redis ||= self.class.build_redis(**redis_options)
+      end
+
+      def inspect
+        instance = @redis || @redis_options
+        "<##{self.class} options=#{options.inspect} redis=#{instance.inspect}>"
+      end
+
+      # Cache Store API implementation.
+      #
+      # Read multiple values at once. Returns a hash of requested keys ->
+      # fetched values.
+      def read_multi(*names)
+        if @mget_capable
+          begin
+            read_multi_mget(*names)
+          rescue Redis::Distributed::CannotDistribute
+            @mget_capable = false
+            super
+          end
+        else
+          super
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Supports Redis KEYS glob patterns:
+      #
+      #   h?llo matches hello, hallo and hxllo
+      #   h*llo matches hllo and heeeello
+      #   h[ae]llo matches hello and hallo, but not hillo
+      #   h[^e]llo matches hallo, hbllo, ... but not hello
+      #   h[a-b]llo matches hallo and hbllo
+      #
+      # Use \ to escape special characters if you want to match them verbatim.
+      #
+      # See https://redis.io/commands/KEYS for more.
+      #
+      # Failsafe: Raises errors.
+      def delete_matched(matcher, options = nil)
+        instrument :delete_matched, matcher do
+          case matcher
+          when String
+            redis.eval DELETE_GLOB_LUA, [], [namespaced_key(matcher, options)]
+          else
+            raise ArgumentError, "Only Redis glob strings are supported: #{matcher.inspect}"
+          end
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Increment a cached value. This method uses the Redis incr atomic
+      # operator and can only be used on values written with the :raw option.
+      # Calling it on a value not stored with :raw will initialize that value
+      # to zero.
+      #
+      # Failsafe: Raises errors.
+      def increment(name, amount = 1, options = nil)
+        instrument :increment, name, amount: amount do
+          redis.incrby normalize_key(name, options), amount
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Decrement a cached value. This method uses the Redis decr atomic
+      # operator and can only be used on values written with the :raw option.
+      # Calling it on a value not stored with :raw will initialize that value
+      # to zero.
+      #
+      # Failsafe: Raises errors.
+      def decrement(name, amount = 1, options = nil)
+        instrument :decrement, name, amount: amount do
+          redis.decrby normalize_key(name, options), amount
+        end
+      end
+
+      # Cache Store API implementation.
+      #
+      # Removes expired entries. Not supported.
+      def cleanup(options = nil)
+        super
+      end
```

Comment:
Included for clarity, demonstrating that the full cache store API is considered and specifically not implemented.

Reply:
Is expiry not supported at all? Or is expiry not used by default, and then when it is, delegated to redis? (Meaning this method would be a no-op, but would be "working")

Reply:
Expiry is natively supported, right.